extensions/amp-story/1.0/variable-service.js

```diff
@@ -75,43 +75,60 @@ export class AmpStoryVariableService {
 
   /** @private */
   initializeListeners_() {
-    this.storeService_.subscribe(StateProperty.PAGE_IDS, (pageIds) => {
-      this.variables_[AnalyticsVariable.STORY_PAGE_COUNT] = pageIds.length;
-    });
+    this.storeService_.subscribe(StateProperty.PAGE_IDS, () =>
+      this.updatePageCountVariables_()
+    );
 
     this.storeService_.subscribe(
       StateProperty.CURRENT_PAGE_ID,
-      (pageId) => {
-        if (!pageId) {
-          return;
-        }
-
-        this.variables_[AnalyticsVariable.STORY_PREVIOUS_PAGE_ID] =
-          this.variables_[AnalyticsVariable.STORY_PAGE_ID];
-
-        this.variables_[AnalyticsVariable.STORY_PAGE_ID] = pageId;
-
-        const pageIndex = /** @type {number} */ (
-          this.storeService_.get(StateProperty.CURRENT_PAGE_INDEX)
-        );
-        this.variables_[AnalyticsVariable.STORY_PAGE_INDEX] = pageIndex;
-
-        const numberOfPages = this.storeService_.get(
-          StateProperty.PAGE_IDS
-        ).length;
-        if (numberOfPages > 0) {
-          if (numberOfPages === 1) {
-            this.variables_[AnalyticsVariable.STORY_PROGRESS] = 0;
-          } else {
-            this.variables_[AnalyticsVariable.STORY_PROGRESS] =
-              pageIndex / (numberOfPages - 1);
-          }
-        }
-      },
+      () => this.updatePageCountVariables_(),
       true /* callToInitialize */
     );
   }
 
+  /** @private */
+  updatePageCountVariables_() {
+    // For analytics purposes: story page count excludes ads.
+    const pageIds = /** @type {Array<string>} */ (
+      this.storeService_.get(StateProperty.PAGE_IDS)
+    );
+
+    this.variables_[AnalyticsVariable.STORY_PAGE_COUNT] = pageIds.filter(
+      (id) => !id.startsWith('i-amphtml-ad-')
+    ).length;
+
+    const pageId = /** @type {string} */ (
+      this.storeService_.get(StateProperty.CURRENT_PAGE_ID)
+    );
+    if (!pageId) {
+      return;
+    }
+
+    this.variables_[AnalyticsVariable.STORY_PREVIOUS_PAGE_ID] =
+      this.variables_[AnalyticsVariable.STORY_PAGE_ID];
+
+    this.variables_[AnalyticsVariable.STORY_PAGE_ID] = pageId;
+
+    const pageIndex = /** @type {number} */ (
+      this.storeService_.get(StateProperty.CURRENT_PAGE_INDEX)
+    );
+    const adsBeforePage = pageIds
+      .slice(0, pageIndex)
+      .filter((id) => id.startsWith('i-amphtml-ad-')).length;
+    this.variables_[AnalyticsVariable.STORY_PAGE_INDEX] =
+      pageIndex - adsBeforePage;
+
+    const numberOfPages = pageIds.length;
```

Comment:
Shouldn't the number of pages here not contain the ad pages? So that the variables don't care about the ad pages.

My suggestion is at the beginning do `pageIds = storeService.get().filter((id) => id.startsWith...)` and then you just work with the pageIds without the ads.

You can also do `if pageId.startsWith('i-amphtml-ad') {return}` so that you don't run this variable update if the current page is an ad